tests/integration/test_lambda.py

```diff
@@ -141,6 +142,26 @@ def test_create_lambda_function(self):
         self.assertEqual(rs['Configuration'].get('VpcConfig', {}), vpc_config)
         self.assertEqual(rs['Tags'], tags)
 
+        # Get funtion by ARN
+        function_arn = rs['Configuration'].get('FunctionArn')
+        rs = client.get_function(
+            FunctionName=function_arn
+        )
+        self.assertEqual(rs['Configuration'].get('KMSKeyArn', ''), kms_key_arn)
+        self.assertEqual(rs['Configuration'].get('VpcConfig', {}), vpc_config)
+        self.assertEqual(rs['Tags'], tags)
+
+        # Get funtion by partial ARN
+        # i.e. us-east-1:000000000000:function:{function_name}
+        partial_function_arn = function_arn.split(':')[3:]
+        partial_function_arn = ':'.join(partial_function_arn)
+        rs = client.get_function(
+            FunctionName=function_arn
```

Comment:
I think we probably want to pass `FunctionName=partial_function_arn` here. Also, to slightly reduce duplication between lines 137-143, 146-152, and 158-164, we could run the 3 sets of assertions in a loop:

```
for func_ref in (func_name, function_arn, partial_function_arn):
    rs = client.get_function(FunctionName=func_ref)
    ...
```

Reply:
Completely right Waldemmar, just pushed the fix, thanks!